### PR TITLE
Adapted mean predictor to use random samples.

### PR DESCRIPTION
**Pull request by jaheba**
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

/cc @lovvge

### CODE REVIEW

**Comment by szha** [comment]
Job PR-239/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-239/3/index.html

File: src/gluonts/model/trivial/mean.py [hunk 1]
```diff
@@ -60,10 +61,20 @@ def __init__(
...
     def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
         for item in dataset:
-            mean = np.mean(item["target"][-self.context_length :])
+            if self.context_length is not None:
+                target = item["target"][-self.context_length :]
+            else:
+                target = item["target"]
+
+            mean = np.nanmean(target)
+            std = np.nanstd(target)
```
Comment (lostella) [inline src/gluonts/model/trivial/mean.py]: Great, thank you :-) I was about to ask for this in the other PR adding this predictor

File: src/gluonts/model/trivial/mean.py [hunk 2]
```diff
@@ -60,10 +61,20 @@ def __init__(
 
     def predict(self, dataset: Dataset, **kwargs) -> Iterator[SampleForecast]:
         for item in dataset:
-            mean = np.mean(item["target"][-self.context_length :])
+            if self.context_length is not None:
+                target = item["target"][-self.context_length :]
```
Comment (lostella) [inline src/gluonts/model/trivial/mean.py]: I like this option

Comment (jaheba) [reply src/gluonts/model/trivial/mean.py]: Makes it easier to derive this Predictor from others :)

File: src/gluonts/model/trivial/mean.py (hunk 1 above)
Comment (jaheba) [reply src/gluonts/model/trivial/mean.py]: @lovvge deserves some credit as well for this!

**Comment by szha** [comment]
Job PR-239/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-239/4/index.html

**Comment by szha** [comment]
Job PR-239/5 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-239/5/index.html

File: src/gluonts/model/trivial/mean.py [hunk 3]
```diff
@@ -25,12 +25,13 @@
...
 from gluonts.model.forecast import Forecast, SampleForecast
 from gluonts.model.predictor import RepresentablePredictor
+from gluonts.support.pandas import frequency_add
 
 
 class MeanPredictor(RepresentablePredictor):
     """
-    A :class:`Predictor` that predicts the mean of the last `context_length`
-    elements of the input target.
+    A :class:`Predictor` that predicts the samples based on the mean of the
```
Comment (lostella) [inline src/gluonts/model/trivial/mean.py]: Based on the mean and standard deviation

**Pull request merged**